### PR TITLE
fix(modal): card modal can now be swiped to close on the content

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue URL: resolves https://github.com/ionic-team/ionic-framework/issues/22046

## What is the new behavior?

- Card modal can now be swiped to close from `ion-content`
- See https://www.notion.so/ionic/Modal-813deb5422fb48a48ff554ef2466163e for a list of when the swiping should/should not happen.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Dev build: 6.1.3-dev.11650647187.10053e0b

### CODE REVIEW

File: core/src/components/modal/gestures/swipe-to-close.ts [hunk 1]
```diff
@@ -24,17 +26,34 @@ export const createSwipeToCloseGesture = (el: HTMLIonModalElement, animation: An
...
+     * If we are swiping on the content,
+     * swiping should only be possible if
+     * the content is scrolled all the way
+     * to the top so that we do not interfere
+     * with scrolling.
+     */
+    const content = target.closest('ion-content');
+    if (content) {
+      const root = getElementRoot(content);
+      const scrollEl = root.querySelector('.inner-scroll') as HTMLElement;
```
Comment (sean-perkins) [inline core/src/components/modal/gestures/swipe-to-close.ts:39]: Are there any considerations here for `.ion-content-scroll-host` and virtual scroll implementations?

File: .github/workflows/actions/update-reference-screenshots/action.yml [hunk 2]
```diff
@@ -13,6 +13,11 @@ runs:
     - uses: actions/download-artifact@v2
       with:
         path: ./artifacts
+    - uses: ./.github/workflows/actions/upload-archive
```
[the remark this answers is not in the document]
Comment (liamdebeasi) [reply .github/workflows/actions/update-reference-screenshots/action.yml:16]: Oops sorry ignore that - I'm debugging an issue with screenshots updating. that will get removed when I fix the issue.

File: core/src/components/modal/gestures/swipe-to-close.ts (hunk 1 above)
Comment (liamdebeasi) [reply core/src/components/modal/gestures/swipe-to-close.ts:39]: 👍 good catch will fix

**Review by sean-perkins — APPROVED** [review]

**Comment by amandaejohnston** [comment]
I'm unable to swipe the modal closed on the new `card-scroll-target` test; should I be able to? I'm testing on both my PC (Windows Chrome 100.0.4896.127) and my phone (Android 12 Chrome). I've tried swiping on both the header and the content.

**Comment by liamdebeasi** [comment]
Are you testing in MD mode or iOS mode? The card modal is only available in iOS mode.

**Comment by amandaejohnston** [comment]
Aha, that explains it 😆 Sorry about that.

File: core/src/components/modal/test/card-scroll-target/modal.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import { expect } from '@playwright/test';
+import { dragElementBy, test } from '@utils/test/playwright';
+
+test.describe('card modal - scroll target', () => {
```
Comment (amandaejohnston) [inline core/src/components/modal/test/card-scroll-target/modal.e2e.ts:4]: Is this still needed? Looks like everything is duplicated in the `card` test.

Comment (liamdebeasi) [reply core/src/components/modal/test/card-scroll-target/modal.e2e.ts:4]: This test duplicates the tests in `card` but with the `.ion-content-scroll-host` element. I am trying to check that we are looking at the correct `scrollTop` values when swiping. For example, when using `.ion-content-scroll-host` the scrollTop value of `ion-content`'s inner scroll element will always be `0`. But in that case, we want to check the `.ion-content-scroll-host` scrollTop instead.

Comment (amandaejohnston) [reply core/src/components/modal/test/card-scroll-target/modal.e2e.ts:4]: Ah, gotcha!

**Review by amandaejohnston — APPROVED** [review]

**Pull request merged**